features/step_definitions/auth_code_steps.rb

```diff
@@ -0,0 +1,32 @@
+o_auth_query_params = %i(
+  redirect_uri=http://localhost:3000
+  response_type=code
+  scope=openid%20read
+  nonce=hello
+  state=hi
+).join("&")
+
+Given /^I send a post request from that client to the code flow authorization endpoint$/ do
+  client_json = JSON.parse(last_response.body)
+  @client_id = client_json['o_auth_application']['client_id']
+  @client_secret = client_json['o_auth_application']['client_secret']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.